src/app/model/migrations/migrations/to0002/MigrationToVersion0002.ts

```diff
@@ -0,0 +1,212 @@
+
+import {
+  RootStateFrom,
+  RootStateTo,
+  StoreProgressPayloadTo,
+} from './MigrationReduxTypes0002.ts';
+import { FlowStateFrom } from './MigrationStorageTypes0002.ts';
+import {
+  getUpdatedReduxState,
+  selectNotCompletedActivities,
+  selectNotCompletedFlows,
+} from './MigrationUtils0002';
+import { QueryDataUtils } from './MigrationUtils0002.ts';
+import {
+  NotCompletedEntitiesFrom,
+  NotCompletedEntitiesTo,
+} from './MigrationUtils0002.ts';
+import {
+  IMigration,
+  MigrationInput,
+  MigrationOutput,
+  Storages,
+} from '../../types';
+import { getStorageRecord } from '../../utils.ts';
+
+export class MigrationToVersion0002 implements IMigration {
+  private queryDataUtils: QueryDataUtils;
+
+  constructor(queryClient: QueryClient) {
+    this.queryDataUtils = new QueryDataUtils(queryClient);
+  }
+
+  private getFlowState = (key: string): FlowStateFrom | null => {
+    return getStorageRecord(Storages.FlowProgress, key);
+  };
+
+  private getFlowRecordKey = (
+    appletId: string,
+    flowId: string | null,
+    eventId: string,
+  ) => {
+    const flowKey = flowId ?? 'default_one_step_flow';
+    return `${flowKey}-${appletId}-${eventId}`;
+  };
+
+  private getUpdatedFlowProgress(
+    progressFlowFrom: NotCompletedEntitiesFrom,
+    entityName: string,
+  ): NotCompletedEntitiesTo {
+    const storeProgressPayloadTo: StoreProgressPayloadTo = {
+      ...progressFlowFrom.payload,
+      entityName: entityName,
+    };
+
+    return {
+      appletId: progressFlowFrom.appletId,
+      eventId: progressFlowFrom.eventId,
+      entityId: progressFlowFrom.entityId,
+      type: progressFlowFrom.type,
+      payload: storeProgressPayloadTo,
+    };
+  }
+
+  private getUpdatedActivityProgress(
+    progressFlowFrom: NotCompletedEntitiesFrom,
+    entityName: string,
+  ): NotCompletedEntitiesTo {
+    const storeProgressPayloadTo: StoreProgressPayloadTo = {
+      ...progressFlowFrom.payload,
+      entityName: entityName,
+    };
+
+    return {
+      appletId: progressFlowFrom.appletId,
+      eventId: progressFlowFrom.eventId,
+      entityId: progressFlowFrom.entityId,
+      type: progressFlowFrom.type,
+      payload: storeProgressPayloadTo,
+    };
+  }
+  private getUpdatedProgressForFlows(
+    reduxState: RootStateFrom,
+  ): NotCompletedEntitiesTo[] {
+    const progressFlowsFrom = selectNotCompletedFlows(reduxState);
+    const progressFlowsTo: NotCompletedEntitiesTo[] = [];
+
+    for (const progressFlowFrom of progressFlowsFrom) {
+      const { appletId, entityId, eventId } = progressFlowFrom;
+
+      const logAppletName = '',
+        logFlowName = '';
+      const logFlowStateFrom = '';
+      const logCurrentActivityDto = '';
+      const logActivityFlowDto = '';
+      const logProgressFlowFrom = JSON.stringify(progressFlowFrom, null, 2);
+
+      try {
+        const key = this.getFlowRecordKey(appletId, entityId, eventId);
+
+        const flowStateFrom = this.getFlowState(key)!;
```

Comment:
Please rename to `flowState` as we don't migrate it

Reply:
@BamMironov  Let's either: leave variable name as is or as alternative - rename variable to flowState and the function to getFlowStateTo ? I'd prefer to see in the code is it source or destination.

Reply:
Okay, let's leave it as-is